src/gt4py/next/program_processors/runners/dace_iterator/__init__.py

```diff
@@ -307,3 +312,177 @@ def build_sdfg_from_itir(
+        )
+        for closure in itir_tmp.closures:  # type: ignore[union-attr]
+            shifts = itir_transforms.trace_shifts.TraceShifts.apply(closure)
+            for k, v in shifts.items():  # type: ignore[assignment]
+                if not isinstance(k, str):
+                    continue
+                if k not in sdfg.gt4py_program_input_fields:
+                    continue
+                sdfg.offset_providers_per_input_field.setdefault(k, []).extend(list(v))
+
+        return sdfg
+
+    def __sdfg_closure__(self, reevaluate: Optional[dict[str, str]] = None) -> dict[str, Any]:
+        """
+        The connectivity tables are defined here symbolically (runtime args and not compile time).
+        They need to be defined in `__sdfg_closure__` due to the fact that they are not part of GT4Py Program's arguments.
+        Actually, they are already defined in the construction of the sdfg, but some symbols are missing because they are not part of GT4Py Program's arguments -parsing issue-.
+        Here, we define them symbolically (along with the correct dtypes), and let DaCe fill the missing symbols.
+        Keep in mind, that __sdfg_closure__ is called after __sdfg__.
+        """
+        offset_provider = self.connectivities
+
+        symbols = {
+            f"__{connectivity_identifier(k)}_{sym}": dace.symbol(
+                f"__{connectivity_identifier(k)}_{sym}"
+            )
+            for k, v in offset_provider.items()  # type: ignore[union-attr]
+            for sym in ["size_0", "size_1", "stride_0", "stride_1"]
+            if hasattr(v, "table")
+            and connectivity_identifier(k) in self.sdfg_closure_vars["sdfg.arrays"]
+        }
+
+        closure_dict = {}
+        for k, v in offset_provider.items():  # type: ignore[union-attr]
+            conn_id = connectivity_identifier(k)
+            if hasattr(v, "table") and conn_id in self.sdfg_closure_vars["sdfg.arrays"]:
+                if conn_id not in Program.connectivity_tables_data_descriptors:
+                    Program.connectivity_tables_data_descriptors[conn_id] = dace.data.Array(
+                        dtype=dace.int64 if v.index_type == np.int64 else dace.int32,
+                        shape=[symbols[f"__{conn_id}_size_0"], symbols[f"__{conn_id}_size_1"]],
+                        strides=[
+                            symbols[f"__{conn_id}_stride_0"],
+                            symbols[f"__{conn_id}_stride_1"],
+                        ],
+                        storage=Program.connectivity_tables_data_descriptors["storage"],
+                    )
+                closure_dict[conn_id] = Program.connectivity_tables_data_descriptors[conn_id]
+
+        return closure_dict
+
+    def __sdfg_signature__(self) -> tuple[Sequence[str], Sequence[str]]:
+        args = []
+        for arg in self.past_stage.past_node.params:
+            args.append(arg.id)
+        return (args, [])
+
+    @classmethod
+    def crosscheck_dace_parsing(
```

Comment:
This shouldn't be a class method but just a private free function in the module.